libsolidity/interface/CompilerStack.cpp

```diff
@@ -507,21 +507,44 @@ string CompilerStack::applyRemapping(string const& _path, string const& _context
 			return false;
 		return std::equal(_a.begin(), _a.end(), _b.begin());
 	};
+	// Try to find whether _a is a closer match for context _reference than _b
+	// Defaults to longest prefix in case of a tie.
+	auto isClosestContext = [](string const& _a, string const& _b, string const& _reference)
+	{
+		int a = _reference.compare(_a);
+		int b = _reference.compare(_b);
+		if (a == 0)
+			return true;
+		else if (b == 0)
+			return false;
+		else if (abs(a) == abs(b)) {
+			return a > 0;
+		}
+		return abs(a) < abs(b);
+	};
 
+	using filepath = boost::filesystem::path;
+	filepath context(_context);
 	size_t longestPrefix = 0;
 	string longestPrefixTarget;
+	string currentClosestContext;
+	string referenceContext = context.parent_path().generic_string();
 	for (auto const& redir: m_remappings)
 	{
+		filepath redirContext(redir.context);
 		// Skip if we already have a closer match.
-		if (longestPrefix > 0 && redir.prefix.length() <= longestPrefix)
+		if (longestPrefix > 0 && redir.prefix.length() < longestPrefix)
```

Comment:
I'm not sure the point you're making here. Are you saying that we should check context first as opposed to checking the prefix length first?